client/fingerprint/plugins_cni.go

```diff
@@ -0,0 +1,67 @@
+package fingerprint
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+const (
+	defaultCNIPath     = "/opt/cni/bin"
+	cniPluginAttribute = "plugins.cni.version"
+)
+
+// PluginsCNIFingerprint creates a fingerprint of the CNI plugins present on the
+// CNI plugin path specified for the Nomad client.
+type PluginsCNIFingerprint struct {
+	StaticFingerprinter
+	logger hclog.Logger
+	lister func(string) ([]os.DirEntry, error)
+}
+
+func NewPluginsCNIFingerprint(logger hclog.Logger) Fingerprint {
+	return &PluginsCNIFingerprint{
+		logger: logger.Named("cni_plugins"),
+		lister: os.ReadDir,
+	}
+}
+
+func (f *PluginsCNIFingerprint) Fingerprint(req *FingerprintRequest, resp *FingerprintResponse) error {
+	cniPath := req.Config.CNIPath
+	if cniPath == "" {
+		cniPath = defaultCNIPath
+	}
+
+	entries, err := f.lister(cniPath)
+	switch {
+	case err != nil:
+		f.logger.Debug("failed to read CNI plugins directory", "cni_path", cniPath, "error", err)
+		resp.Detected = false
+		return nil
+	case len(entries) == 0:
+		f.logger.Debug("no CNI plugins found", "cni_path", cniPath)
+		resp.Detected = true
+		return nil
+	}
+
+	bridgePath := filepath.Join(cniPath, "bridge")
```

Comment:
Would it make sense to fingerprint each individual plugin separately? For Nomad `bridge` may be enough, but for environments that use custom CNI plugins it could be helpful to have them fingerprinted as well.

Reply:
Yeah we can do that; most likely they'll all be the same version (extracted from the same archive) but at least folks will have a view of what's there

Reply:
Cool! Yeah, let's hope they are all the same 😅 

But for a custom plugin I would imagine they would have their own version.